search-service/src/main/kotlin/com/egm/stellio/search/service/EntityPayloadService.kt

```diff
@@ -103,6 +103,47 @@ class EntityPayloadService(
             .bind("specific_access_policy", specificAccessPolicy?.toString())
             .execute()
 
+    @Transactional
+    suspend fun mergeEntity(
+        entityId: URI,
+        ngsiLdAttributes: List<NgsiLdAttribute>,
+        expandedAttributes: ExpandedAttributes,
+        observedAt: ZonedDateTime?,
+        sub: Sub?
+    ): Either<APIException, UpdateResult> = either {
+        val createdAt = ZonedDateTime.now(ZoneOffset.UTC)
+        logger.debug("Merging entity {}", entityId)
+
+        val temporalEntityAttributes = temporalEntityAttributeService.getForEntity(entityId, emptySet())
+        temporalEntityAttributes.filter {
+            !ngsiLdAttributes.map { it.name }.contains(it.attributeName)
+        }.forEach {
+            temporalEntityAttributeService.deleteTemporalAttribute(
```

Comment:
I don't get why you are deleting attributes

Reply:
the description in the spec says that `This operation allows modification of an existing NGSI-LD Entity aligning to the JSON Merge Patch processing rules defined in IETF RFC 7396 [16] by adding new Attributes (Properties or Relationships) or modifying or deleting existing Attributes associated with an existing Entity.`. So for me, I understood that it is necessary to add attributes when they do not exist, update them when they exist and delete them if they are no longer present

Reply:
the removal is done via the use of NGSI-LD Null (see 5.6.17 - Merge Entity), that we'll implement later